### DISCUSSION TITLE
Add condition to not take new trade when we take profit or hit a stop loss

### DISCUSSION THREAD

**Discussion by abeerzee** [kernc/backtesting.py#660] [General]
Greetings all,

I just started out my journey into backtesting and python programming as such recently. I was able to implement my strategy, however there was one thing I was struggling with i.e. to code a condition such that I want to avoid taking a new trade in the same candle where we have hit a stop loss or have taken profit. Below is some is one of sample condition I'm checking before going long

        if (  self.data.Close[-2] > self.sma1[-2]): 
            if(self.data.Close[-1] > self.data.Close[-2]  and not self.position.is_long):
                self.buy(tp=1.1*price, sl=.985*price,size=50)   

Any idea how do I add this condition for this in second if statement ? I want to go long only after closing of 2 candles after the  candle where we hit SL or take profit. I saw some variables like exit_bar in closed trades which might help but i'm not able to use them.

**Comment by kernc** [comment]
You can relate [`Trade.exit_bar`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Trade.exit_bar) to current bar by the length of [`Strategy.data`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.data) at each iteration. Say:
```py
    ...
    last_trade = self.closed_trades[-1]
    if len(self.data.index) - last_trade.exit_bar > 2:
        # At least two days since last trade was closed
        self.buy(...)

**Reply by abeerzee** [reply]
Thanks a lot, I was able to make it work :)
